pkgs/development/tools/parsing/antlr/4.nix

```
@@ -0,0 +1,157 @@
+{ lib, stdenv, fetchurl, jre
+, fetchFromGitHub, cmake, ninja, pkg-config, darwin
+
+# ANTLR 4.8 & 4.9
+, libuuid
+
+# ANTLR 4.9
+, utf8cpp }:
+
+let
+
+  mkAntlr = { version, sourceSha256, jarSha256 }: lib.fixedPoints.makeExtensible (self: {
+    source = fetchFromGitHub {
+      owner = "antlr";
+      repo = "antlr4";
+      rev = version;
+      sha256 = sourceSha256;
+    };
+
+    antlr = stdenv.mkDerivation {
+      pname = "antlr";
+      inherit version;
+
+      src = fetchurl {
+        url = "https://www.antlr.org/download/antlr-${version}-complete.jar";
+        sha256 = jarSha256;
+      };
+
+      dontUnpack = true;
+
+      installPhase = ''
+        mkdir -p "$out"/{share/java,bin}
+        cp "$src" "$out/share/java/antlr-${version}-complete.jar"
+
+        echo "#! ${stdenv.shell}" >> "$out/bin/antlr"
+        echo "'${jre}/bin/java' -cp '$out/share/java/antlr-${version}-complete.jar:$CLASSPATH' -Xmx500M org.antlr.v4.Tool \"\$@\"" >> "$out/bin/antlr"
+
+        echo "#! ${stdenv.shell}" >> "$out/bin/grun"
+        echo "'${jre}/bin/java' -cp '$out/share/java/antlr-${version}-complete.jar:$CLASSPATH' org.antlr.v4.gui.TestRig \"\$@\"" >> "$out/bin/grun"
+
+        chmod a+x "$out/bin/antlr" "$out/bin/grun"
+        ln -s "$out/bin/antlr"{,4}
+      '';
+
+      inherit jre;
+
+      passthru = {
+        inherit (self) runtime;
+        jarLocation = "${self.antlr}/share/java/antlr-${version}-complete.jar";
+      };
+
+      meta = with lib; {
+        description = "Powerful parser generator";
+        longDescription = ''
+          ANTLR (ANother Tool for Language Recognition) is a powerful parser
+          generator for reading, processing, executing, or translating structured
+          text or binary files. It's widely used to build languages, tools, and
+          frameworks. From a grammar, ANTLR generates a parser that can build and
+          walk parse trees.
+        '';
+        homepage = "https://www.antlr.org/";
+        sourceProvenance = with sourceTypes; [ binaryBytecode ];
+        license = licenses.bsd3;
+        platforms = platforms.unix;
+      };
+    };
+
+    runtime = {
+      cpp = stdenv.mkDerivation {
+        pname = "antlr-runtime-cpp";
+        inherit version;
+        src = self.source;
+
+        outputs = [ "out" "dev" "doc" ];
+
+        nativeBuildInputs = [ cmake ninja pkg-config ];
+        buildInputs =
+          lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.CoreFoundation;
```

```suggestion
        buildInputs = lib.optional stdenv.isDarwin CoreFoundation;
```
normally we inherit this in top-level.